decidim-proposals/app/controllers/concerns/decidim/proposals/needs_permission.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  module Proposals
+    # Common logic to ordering resources
+    module NeedsPermission
+      extend ActiveSupport::Concern
+
+      included do
+        helper_method :allowed_to?
+
+        class ::Decidim::MissingPermission < StandardError
+        end
+
+        rescue_from Decidim::MissingPermission, with: :user_not_authorized
+
+        def check_permission_to(action, subject, extra_context = {})
+          raise Decidim::MissingPermission unless allowed_to?(action, subject, extra_context)
```

Comment:
This exception is misleading. The permission isn't missing, but instead youre forbidden to do that action. Something like `Decidim::ActionForbidden` or `Decidim::Permissions::Forbidden` sounds better.

Reply:
Done!